src/templater.rs

```diff
@@ -583,24 +584,39 @@ impl CommitOrChangeId {
     }
 
     pub fn short_prefix_and_brackets(&self, repo: RepoRef) -> String {
-        highlight_shortest_prefix(self, 12, repo)
-    }
-}
-
-fn highlight_shortest_prefix(id: &CommitOrChangeId, total_len: usize, repo: RepoRef) -> String {
-    let prefix_len = repo
-        .base_repo()
-        .shortest_unique_prefix_length(id.as_bytes());
-    let mut hex = id.hex();
-    if prefix_len < total_len - 2 {
-        format!(
-            "{}[{}]",
-            &hex[0..prefix_len],
-            &hex[prefix_len..total_len - 2]
-        )
+        highlight_shortest_prefix_brackets(self, 12, repo)
+    }
+}
+
+/// This function supports short `total_len` by ensuring that the entire
+/// unique prefix is always printed
+fn extract_entire_prefix_and_trimmed_tail(
+    s: String,
+    prefix_len: usize,
+    total_len: usize,
+) -> (String, String) {
+    let mut prefix = s;
+    // Prefix is not limited by `total_len`
+    let mut rest = prefix.split_off(min(prefix_len, prefix.len()));
+    rest.truncate(total_len - 2 - min(total_len - 2, prefix_len));
```

Comment:
Why do -2 here for the non-bracket case? I understand the total_len can be adjusted, but I think it might be confusing (compared to maybe +2 in highlight_shortest_prefix_brackets).

Reply:
You are right, it's a mistake. I thought the grey tails looked shorter than I expected.